### ISSUE TITLE
Add Project-Based Rule

### ISSUE THREAD

**Issue by mikeendale** [blencorp/rules-gen#7]
### Tasks
- [ ] **Tech Stack Detection**: Analyze project files to identify the tech stack.
  - Check `package.json` for dependencies (e.g., `"next"`, `"typescript"`).
  - Look for configuration files like `tsconfig.json` or `next.config.js`.
- [ ] **Rule Mappings**: Define pre-selected rule sets for common stacks.
  - Example: Next.js + TypeScript rules (e.g., type safety, Next.js routing conventions).
- [ ] **Initial Prompt**: Add a mouse-navigable prompt at the CLI start:
  ```
  How would you like to select rules?
  > Based on your project (Detected: Next.js with TypeScript)
    Start fresh with manual selection
  [Use mouse up/down to select, Enter to confirm]
  ```
- [ ] **Confirmation Step**: If a stack is detected, prompt for confirmation:
  ```
  Detected Next.js with TypeScript. Use these pre-selected rules? [Y/n]
  ```
- [ ] **Fallback Options**: Handle cases where no stack is detected:
  ```
  Couldn't detect your tech stack. How would you like to proceed?
  > Choose a stack manually (e.g., Next.js, React, Django)
    Start fresh with no pre-selection
  [Use mouse up/down to select, Enter to confirm]
  ```
- [ ] **Customization**: Allow users to refine pre-selected rules:
  ```
  Rules for Next.js + TypeScript selected. Want to add or remove rules? [Y/n]
  ```
- [ ] **CLI Flags**: Add options for advanced users:
  - `--detect-stack`: Automatically apply project-based rules without prompting.
  - `--fresh`: Skip detection and start fresh.
- [ ] **Documentation**: Update README with examples and usage instructions.

### Acceptance Criteria
- The CLI correctly identifies common tech stacks (e.g., Next.js with TypeScript) from project files.
- Users can navigate all prompts using **mouse up and down** (scrolling or clicking) and confirm selections with Enter.
- Pre-selected rules match the detected tech stack and are applied accurately.
- Users can confirm, customize, or override pre-selected rules as needed.
- Fallback options are available and functional when detection fails.
- CLI flags (`--detect-stack`, `--fresh`) work as intended.

### Notes
- **Mouse Support**: Implement using a library that enables mouse interactions in the terminal (e.g., a modified `inquirer.js` or a TUI framework like `blessed`). Ensure compatibility with common terminal emulators.
- **Flexibility**: Design detection and rule mappings to be extensible for additional stacks (e.g., React, Django) in the future.
- **User Experience**: Make mouse navigation intuitive, supporting scroll events or clickable list items.
- **Error Handling**: Display clear messages for detection failures or unsupported interactions.
- **Initial Stacks**: Focus on Next.js + TypeScript, React, and Django as starting points.

### Priority
High

**Comment by mikeendale** [comment]
Done.
